recipes/boost/all/conanfile.py

```diff
@@ -1567,7 +1569,8 @@ def filter_transform_module_libraries(names):
                     self.cpp_info.components["stacktrace_backtrace"].defines.append("BOOST_STACKTRACE_USE_BACKTRACE")
                     self.cpp_info.components["stacktrace_backtrace"].requires.append("libbacktrace::libbacktrace")
 
-                self.cpp_info.components["stacktrace_noop"].defines.append("BOOST_STACKTRACE_USE_NOOP")
```

Comment:
stacktrace_noop is a header-only library.
So the component is always available (when stacktrace is enabled)

Sure, you can add an option to specifically disable it, but is it worth it?

If you push further, please also do the following:
- in `configure`, do `del self.options.with_stcktrace_noop` if stacktrace is disabled.
- Also `do `del self.info.options.with_stacktrace_noop`  in `package_id` (this can be done unconditionally)

Reply:
What do you mean stacktrace_noop is a header-only library? From what I understand, the noop is an implementation detail that is activated with the `BOOST_STACKTRACE_USE_NOOP` define. I'm new to the conan package, so is there another way to disable the noop define? Thanks!

Reply:
Look in the main recipe, there is only one `noop`  match:
https://github.com/conan-io/conan-center-index/blob/4f4a376da214b9ff574c073060045e9ae8994edd/recipes/boost/all/conanfile.py#L1570

This component is set when stacktrace is enabled:
https://github.com/conan-io/conan-center-index/blob/4f4a376da214b9ff574c073060045e9ae8994edd/recipes/boost/all/conanfile.py#L1552

When writing the stacktrace tests, I found out that noop is the fallback backend of stacktrace.
If no implementation can be found, then nothing (=no-op) will be done.

The only way to disable no-op, is to make sure that another backend is available and used.

Reply:
I see, then shouldn't the backtrace backend (which is `True` by default) disable no-op?

https://github.com/conan-io/conan-center-index/blob/4f4a376da214b9ff574c073060045e9ae8994edd/recipes/boost/all/conanfile.py#L136

Reply:
No? no-op is always available, even when the addr2line/libbacktrace backend is built.

All three boost_stacktrace backends (noop/addr2line/libbacktrace) can be available in the same pacakge, just not used in the same target when using the cmake targets.

You can switch between the backends by carefully setting the definitions and linking to the correct libraries.

Reply:
Appreciate the clarification. I'm still a little unsure as to how to ensure no-op isn't used. You mention setting definitions and linking the correct libraries. I have the below CMakeLists.txt for a test. What would I set here to guarantee no-op isn't used? 

```
cmake_minimum_required(VERSION 3.16)
project(stacktrace-test)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup()

add_executable(main main.cpp)
target_link_libraries(main ${CONAN_LIBS})
```

Reply:
What is listed in your conan recipe?
You're only using `conanbuildinfo.cmake`, generated by the `cmake` generator.
If `boost/1.76.0` is in your requirements, you will link to **ALL** of boost.
Not only all of boost stacktrace, but also boost.test, boost.math, boost.filesystem, boost.log, ...

To guarantee you're using noop, you need the following recipe:
```cmake
cmake_minimum_required(VERSION 3.16)
project(stacktrace-test)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(Boost REQUIRED)

add_executable(main main.cpp)
target_link_libraries(main PRIVATE Boost::stacktrace_noop)
```

Reply:
How do I make sure I'm **not** using no-op? 

This is my `conanfile.txt`:

```
[requires]
boost/1.77.0

[generators]
cmake
```

Thanks for the help!

Reply:
> How do I make sure I'm **not** using no-op?

Oh, little nuance :smile: 

> This is my `conanfile.txt`:
> 
> ```
> [requires]
> boost/1.77.0
> 
> [generators]
> cmake

Add `cmake_find_package` to generate a `FindBoost.cmake` file.

And do the following in your cmake script:
```cmake
cmake_minimum_required(VERSION 3.16)
project(stacktrace-test)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(Boost REQUIRED)

add_executable(main main.cpp)
target_link_libraries(main PRIVATE Boost::filesystem)  # any boost target, or none. Just don't add stacktrace_noop
```

`${CONAN_LIBS}` contains **ALL** libraries of all your dependencies.
Using `TARGETS` to the `conan_basic_setup` call is also important.
When not using it, it will add add defines + include dirs of all components.
That includes the `BOOST_STACKTRACE_USE_NOOP` define which is probably confusing you.


> ```
> 
> Thanks for the help!

Reply:
Thanks, got it working!

This was the `CMakeLists.txt` that I ended up with:

```
cmake_minimum_required(VERSION 3.16)
project(stacktrace-test)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

include(${CMAKE_BINARY_DIR}/Findlibbacktrace.cmake)
include(${CMAKE_BINARY_DIR}/FindBoost.cmake)


find_package(Boost REQUIRED)

add_executable(main main.cpp)
target_link_libraries(main PRIVATE Boost::stacktrace_backtrace)
```

Thank you again for working through this with me! It's much appreciated. 

I will withdraw this PR and close the issue as well.

Reply:
Great, please read the documentation of [find_package](https://cmake.org/cmake/help/latest/command/find_package.html) because the include you're using is weird (=not standard cmake)

Reply:
Will do, thanks!

Also, you mentioned that `cmake` would be deprecated as a generator. Is that for `conan` as a whole or just `boost`?

Reply:
I mean only for Conan.
`conanbuiodinfo.cmake` will be a thing of the past.
I'm taking about Conan 2.0, coming in the (near) future.

Reply:
I see, thanks!

As an aside, what build system would you recommend outside of `cmake` these days?

Reply:
To be honest, I use cmake for all of my private and professional c/c++ projects.
All other build systems have disadvantages that are showstoppers for me.

Though, a simple Makefile might be useful for something that has nothing to do with compiling sources.

[Meson}(https://mesonbuild.com/) is used by some OSS projects.
But I find its documentation hard to read + it is opiniated.
It has a potential though.
A meson syntax with a cmake backend would be very nice..

The grandfather of build systems for OSS is of course autotools (autoconf + automake + libtool).
But that is geared mostly towards Linux + gcc. It can work with MSVC on Windows, but it's harder.

If you really don't care about portability, you can also of course just use Visual Studio projects (.snl and .vcxproj files).

So in general, I can't really recommend anything but cmake.

Reply:
Thank you, appreciate your insights!